### PR TITLE
PCHR-2476: Add balance report template

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR adds the template for the Leave & Absence's balance report.

## Style Guide
screenshot 2017-08-07 15 17 05

### SSP
image

### Admin Portal
image

## Comments
* No GIFs needed, these templates are static.
* The template is based on this page of the style guide:
[https://projects.invisionapp.com/d/main#/console/4380186/148828068/preview](https://projects.invisionapp.com/d/main#/console/4380186/148828068/preview)
* This PR is closely related to: [https://github.com/compucorp/civihr-employee-portal-theme/pull/198](https://github.com/compucorp/civihr-employee-portal-theme/pull/198)
* BackstopJS tests were added for both SSP and Admin Portal.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/admin-dashboard/modules/config.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ define([
           .state('calendar', {
             url: '/calendar',
             template: '<manager-leave-calendar contact-id="$root.settings.contactId"></manager-leave-calendar>'
+          })
+          .state('balance-report', {
+            url: '/balance-report',
+            template: '<balance-report></balance-report>'
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/admin-dashboard/modules/config.js]: Should we call it `leave-balance-report` instead? cc @AkA84

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/balance-report.html [hunk 2]
```diff
@@ -0,0 +1,100 @@
+<div class="chr_balance-report">
+  <!-- START:.filters -->
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/balance-report.html]: Please remove comments (same for other comments in this file).

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/balance-report.html [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  <div class="chr_balance-report__filters panel panel-default">
+    <header class="panel-heading panel-heading-plain">
+      <h3 class="panel-title">
+        <i class="fa fa-chevron-down" aria-hidden="true"></i>
+        Show Filters
+      </h3>
+    </header>
+
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/balance-report.html]: No need to add those empty lines I think. Please remove them (all in this file)

**Comment by igorpavlov** [comment]
> comments: No GIFs, these templates don't actually do anything.

to

> No GIFs needed, these templates are static.

**Comment by igorpavlov** [comment]
The styling looks too different in SSP and Admin. Could you please provide a link to the original style mockup?

**Comment by igorpavlov** [comment]
Please change

> These two PRs are closely related

to 

> This PR is closely related to

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/balance-report.html (hunk 3 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/balance-report.html]: Please also add Backstop tests as we discussed.

**Comment by igorpavlov** [comment]
Please also add Backstop tests as we discussed.

**Comment by reneolivo** [comment]
@igorpavlov the differences are caused by the Admin Portal overriding the styles applied to the report. I could force the Admin Portal to render the report as the style guide, but I'm not sure if this was intended. I left a note on the ticket asking for comments.

**Comment by igorpavlov** [comment]
Please move SSP backstop to SSP folder. Also could you please rename backstop to leave-balance-report.js?

**Pull request merged**